crates/gpui/src/platform/linux/wayland/client.rs

```diff
@@ -412,6 +432,9 @@ impl Dispatch<wl_keyboard::WlKeyboard, ()> for WaylandClientState {
                                     },
                                     is_held: false, // todo!(linux)
                                 }));
+
+                                // Need to implement state.repeat_delay and state.repeat_rate
+                                // todo!(linux)
```

Comment:
Key repeat must be handled by the wayland client.

I'd do it myself but I don't understand async rust and the executor well enough. IIUC, to schedule a task to run after a timeout I should do something like this? Or should I cancel the task in the key-release event? If so, how do I cancel a task?

```rust
state.platform_inner.foreground_executor.spawn(async move {
  Timer::after(state.repeat_delay).await;
  if (!was_canceled) {
    trigger_keypress_repeat();
  }
})
```